rpm/build/SPECS/jenkins.spec

```diff
@@ -76,30 +76,57 @@ rm -rf "%{buildroot}"
 /usr/sbin/useradd -g %{name} -s /bin/false -r -c "@@SUMMARY@@" \
 	-d "%{workdir}" %{name} &>/dev/null || :
 
+  # Used to decide later if we should perform a chown in case JENKINS_INSTALL_SKIP_CHOWN is false
+  # Check if a previous installation exists, if so check the JENKINS_HOME value and existing owners of work, log and cache dir, need to to this check
+  # here because the %files directive overwrites folder owners, I have not found a simple way to make the
+  # files directive to use JENKINS_USER as owner.
+  if [ -f "/etc/sysconfig/%{name}" ]; then
+      . /etc/sysconfig/%{name}
+      touch /tmp/cacheowner
```

Comment:
No need to touch if you write to it in the following line.

Also, files should be named such that it's trivial to understand for admins who/what created them, e.g. if the installer fails to finish and the files are left behind. Also, file name collisions: Maybe append the current process ID (`$$`) or similar to prevent that? (Not sure if that works for installer scripts though)